network-libp2p/src/config.rs

```diff
@@ -27,6 +27,7 @@ pub struct Config {
     pub memory_transport: bool,
     pub required_services: Services,
     pub tls: Option<TlsConfig>,
+    pub desired_peer_count: usize,
```

Comment:
```suggestion
    pub desired_peer_count: u32,
```
I don't see a good reason to make the size of this integer platform-specific, the desired peer count doesn't have anything to do with whether the processor is 32 bit or 64 bit.

Reply:
`u8` or `u16` should be enough though.

Reply:
Yep, at least `u16` should be enough.

I think Rust and Google coding guidelines say that you should mostly use 32-bit and 64-bit integers though, and don't think too hard about smaller sizes.

Reply:
@jsdanielh and I discussed this offline and figured out that this is too little of an issue in practice to deal with changing the type.